### ISSUE TITLE
error MSB4006: There is a circular dependency in the target dependency graph involving target "ComputeGetResolvedWinMD"

### ISSUE THREAD

**Issue by mcooley** [microsoft/cppwinrt#950]
Building my project sometimes fails with error MSB4006. The error occurs when there is a Windows Application Packaging project referencing some projects which use the cppwinrt NuGet package, and MSBuild is building the projects in parallel.

I've included [a minimal repro project here](https://microsoft-my.sharepoint-df.com/:u:/p/macool/EYTlF_p-ad5PuTuG7UB5Lz0BElDPNzCDb-aECY458eG8oQ?e=TcqHTe). To reproduce the issue, run `msbuild App1.sln /p:Configuration=Release /p:Platform=x64 /m:4`.

I confirmed that the first version of the NuGet package which exhibits the bug is 2.0.191202.2, so the issue has been present since since #432 was introduced.

**Comment by kennykerr** [comment]
@jlaanstra may have some thoughts but generally the msbuild support is frozen. You can workaround issues in your configuration by forking.

**Comment by jlaanstra** [comment]
@mcooley I've extracted the repro, and ran the above command, but I am not seeing MSB4006.

**Comment by ocalvo** [comment]
I am able to consistently reproduce this problem.

**Comment by chrisglein** [comment]
> I am able to consistently reproduce this problem.

Oh, hi Oscar!
I also am able to reliably repro this. See some detail [here](https://github.com/microsoft/react-native-windows/issues/13599), but in the context of my app project I hit this regularly.

@kennykerr / @jlaanstra, was this resolved as more of a Won't Fix or was this awaiting a repro?

**Comment by JohnMcPMS** [comment]
The target is called `CppWinRTComputeGetResolvedWinMD` in newer versions.  Given that it is conditioned on `CppWinRTGenerateWindowsMetadata` being empty, defining `CppWinRTGenerateWindowsMetadata` to true/false in your project may break the chain.  I'm still verifying it as this is currently affecting our ADO pipelines.
https://github.com/microsoft/cppwinrt/blob/1ccb7e1494b9c55f45f2a306d9a4dbc62317dc4c/nuget/Microsoft.Windows.CppWinRT.targets#L216-L222

**Comment by vineethkuttan** [comment]
> verifying it as this is currently affecting our ADO pipelines.

Does it affect your pipeline @JohnMcPMS ?
